hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/schema/SchemaReplicator.java

```diff
@@ -168,12 +171,21 @@ private boolean isSchemaReplicated(long schemaId) {
      *
      * @param schemas to replicate.
      */
-    public CompletableFuture<Void> replicateAll(List<Schema> schemas) {
-        CompletableFuture[] replications = schemas.stream()
+    public InternalCompletableFuture<Void> replicateAll(List<Schema> schemas) {
+        InternalCompletableFuture[] replications = schemas.stream()
                 .map(this::replicate)
-                .toArray(CompletableFuture[]::new);
-
-        return CompletableFuture.allOf(replications);
+                .toArray(InternalCompletableFuture[]::new);
+
+        InternalCompletableFuture<Void> future = new InternalCompletableFuture<>();
+        CompletableFuture.allOf(replications)
+                .whenCompleteAsync((result, throwable) -> {
+                    if (throwable == null) {
+                        future.complete(null);
+                    } else {
+                        future.completeExceptionally(throwable);
+                    }
+                }, ConcurrencyUtil.getDefaultAsyncExecutor());
+        return future;
```

Comment:
```suggestion
        return CompletableFuture.allOf(replications);
```

Reply:
This doesn't work because the returned future is CompletableFuture. We need ICF to make sure that the callbacks added to the returned future are not executed in some random threads